core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -195,11 +192,19 @@ private void checkApplicationUpdateStatus(Application application) {
                 versionCode = packageInfo.versionCode;
             }
             RudderLogger.logDebug("Current Installed Version: " + versionCode);
+            boolean isLifeCycleEventsAllowed = config.isTrackLifecycleEvents() || config.isRecordScreenViews();
+            if (isLifeCycleEventsAllowed) {
+                application.registerActivityLifecycleCallbacks(this);
+            }
 
             if (previousVersionCode == -1) {
-                // application was not installed previously, Application Installed events
-                RudderLogger.logDebug("Tracking Application Installed");
+                // application was not installed previously, Application Installed event
                 preferenceManager.saveBuildVersionCode(versionCode);
+                // If trackLifeCycleEvents is not allowed then discard the event
+                if (!isLifeCycleEventsAllowed) {
```

Comment:
Can we also check if user opted out before sending `Application Installed` event as well ?

Reply:
Application Installed event will be triggered even before `optOut()` can be called. As optOut() feature depend upon EventRepository object. Please let me know, if I need to check more on this.

Reply:
Okay got it